add dark theme support with @media queries

godalming123 opened the pull request:
This branch works however a few elements still need deferent colors for dark mode.
### elements to add colors for:
- .downloads ul a:active
- a load of colors for code blocks
### elements that I have added colors for:
- body color when you are printing
- `small` element
- blockquete
- dt
- `small` inside `a`
- kbd background
- kbd border
- kbd border-bottom-color
- kbd boxshadow
- kbd color
- `a` hovered
- `pre` color, border
- th
- `a` element
- ul.downloads background
- ul.downloads border
- .downloads li border-right
- .downlads a
- .downloads focus+hover
- th, td border-bottom


I have used `@media (prefers-color-scheme){  }` queries to make this possible https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme. if you would like to see what I have done please check out https://godalming123.github.io/minimal/ the dark theme applies if your device is in dark mode and your browser supports @media prefers color scheme.

godalming123 committed:
add a file for colours

godalming123 committed:
make colors acurate to theme

godalming123 committed:
update

godalming123 committed:
update

godalming123 committed:
improve formatting

godalming123 committed:
add a separate luminosity and color for code text

godalming123 committed:
update colours to use my colours which are based on system theme

godalming123 committed:
make dark the deafualt theme 

this is for if a users device does not support changing theme based on @media querys

godalming123 commented:
is there actually anyone moniterring these projects? I seem to be getting no response

godalming123 commented:
I am struggling to deploy the updated version of this site on git-hub pages. the erreor message is:
![image](https://user-images.githubusercontent.com/68993177/130031521-42d54fa3-3dcc-47dd-acfe-bec7957dc750.png)
I think the issue is because github is expecting me to write
```
body
    //styling
```
and I wrote
```
body {
   //styling
}
```
in the file /_sass/jekyll-theme-minimal.scss line 5
How do you get github to expect the second

godalming123 committed:
use :root instead of body for variables

godalming123 committed:
fix bug

godalming123 committed:
fix bug

godalming123 committed:
fix bug

godalming123 committed:
fix bug

godalming123 commented:
fixed bugs. Now github pages says that my page has passed the checks and is published but I cannot see anything at the url. Is this just github pages being slow at adding it or is there something wrong with the way I'm doing it. Pages says:
![image](https://user-images.githubusercontent.com/68993177/130038781-16dd9fe7-b259-43f5-8f16-cd4240247699.png)

godalming123 commented:
update: has pushed to pages no need to see

godalming123 committed:
use :root instead of body for variables

godalming123 committed:
fix bug

godalming123 committed:
add separate color fo bold text

godalming123 committed:
add separate color for bold text

godalming123 committed:
use h1 color for bold text

godalming123 committed:
use h1 color for bold as well

godalming123 committed:
fix bold text

godalming123 committed:
mention support for dark or light theme in example

godalming123 committed:
reformat + fix theme not changing glitch

godalming123 committed:
fix bug

godalming123 committed:
add theme change transition

godalming123 committed:
add color for table headers

godalming123 committed:
use color for table header

godalming123 committed:
use color for hovered a's

godalming123 committed:
add a color for hovered a's

godalming123 committed:
highlight the fact that theme supports dark mode

godalming123 committed:
mention dark mode support in readme

godalming123 committed:
add more colors for code

godalming123 committed:
use more coding colors

godalming123 committed:
add color for buttons im main section

godalming123 committed:
style more colors for code blocks

godalming123 committed:
style more colors with code colors

godalming123 committed:
fix bug

godalming123 committed:
fix colors

godalming123 committed:
fix color bug where code blocks weren't rounded

godalming123 committed:
fix bug

godalming123 committed:
fix bug

godalming123 committed:
fix bug

godalming123 committed:
fix more colors for code blocks

godalming123 committed:
use separtate color for a tag

BDHU commented:
Sweet. I'll test it out to see how it looks.

godalming123 committed:
add colors for main buttons

godalming123 committed:
add separate colors for buttons

godalming123 committed:
finish making buttons colors

godalming123 committed:
fix bugs

godalming123 committed:
fix bugs

godalming123 committed:
add color for dark hovered main buttons

godalming123 commented:
I think the main issues have been fixed just a bit left to go!
@BDHU I use linux!

godalming123 commented:
oewl and if you use linux @BDHU then you can't use chrome as that heap of a browser doesnt support `@media (prefers-color-theme)` on linux I think its something to do with it not being able to recignise that linux actually allows you to use any theme instead of ugly presets

godalming123 commented:
@BDHU what do you think?

BDHU commented:
My only suggestion is could the horizontal rule be less bright? Currently it's as bright as sunshine. A better alternative is to keep it subtle (if you turn on Dark Reader the horizontal rule is less bright).

Another small thing is can the font color for links be slightly brighter in dark mode (maybe normal mode as well)? Overall, this looks solid.

godalming123 commented:
yes I would agree I was planning to style these I will start work now!

godalming123 committed:
increase brightness of a tags especially in dark mode

godalming123 committed:
add color for table item splitter

godalming123 committed:
use color for table splitter

godalming123 commented:
have implemented all of your suggestions @BDHU

godalming123 committed:
add color for hr

godalming123 commented:
oewl I didnt relise you meant the horizontal rule I thought you meant the splitter for deferent rows in a table anyway I have styled both of them with a separate color for dark mode now.

godalming123 committed:
add demonstration of kbd element on index.md file so I can style it properly

godalming123 committed:
color for kbd elements

godalming123 committed:
make border-bottom darker for kbd elements in dark mode

godalming123 committed:
fix bugs

godalming123 committed:
add color for small element inside a element

godalming123 committed:
fix bugs

godalming123 committed:
add support for dt element

godalming123 committed:
add support for blockqoute

godalming123 committed:
add color for section element

godalming123 committed:
actually theme section

godalming123 committed:
make variable names clearer

godalming123 committed:
make colors clearer

godalming123 committed:
use @mixins for color styles and only apply dark color style if media is on screen so printings arent dark

godalming123 committed:
fix bugs

godalming123 committed:
try to fix bug

godalming123 commented:
I think the major bugs are fixed now I just have to add all the colors for deferent code blocks! @parkr what would I have to do to get this merged.

godalming123 committed:
add coding colors

BDHU commented:
> I think the major bugs are fixed now I just have to add all the colors for deferent code blocks! @parkr what would I have to do to get this merged.

I guess one thing is to add options to disable the dark mode (maybe in config) in case someone doesn't want it.

godalming123 commented:
yeah i dont relly know how to do that i'm still learning jekyll

godalming123 committed:
add code colors + fix github glitch

godalming123 committed:
init

godalming123 commented:
I have an idea as for how to implement being able to change the theme in the config.yaml file but I am very new to jekyll so I do not know how to use variables from it in the deafualt layout file I have created a branch to try and sort this out https://github.com/godalming123/minimal/tree/support-for-changing-theme-in-_config.yaml the idea is theres a **colors-_your theme_** file (_your theme_ can be auto, dark and light) in **/assets/css/** and then we import that file based on a config.yaml option and the contents of the file is just
**/assets/colors-auto.scss**
```
@use "colors";

:root {
    @include colors.colors;
    @include colors.light-colors;
  }
    
@media screen and (prefers-color-scheme: dark) {//we would not like to apply dark mode if content is on printer as that would use a lot of ink
  :root{
    @include colors.dark-colors;
  }
}
```
**/assets/css/colors-dark.scss**
```
@use "colors";

:root {
  @include colors.colors;
  @include colors.dark-colors;
}
```
**/assets/css/colors-light-scss**
```

@use "colors";

:root {
  @include colors.colors;
  @include colors.light-colors;
}
```
However I don't know how to get values of variable from the config.yaml file. Any help?

godalming123 committed:
remove dark-colors file

remove dark-colors file to change theme in config.yaml see https://github.com/godalming123/minimal/tree/support-for-changing-theme-in-_config.yaml if you would like to get that.

godalming123 committed:
reindent

godalming123 committed:
try to add themeing from config.yaml

godalming123 committed:
fix bug

godalming123 committed:
fix bugs

godalming123 committed:
fix bugs so option deafualts to auto

godalming123 committed:
fix bugs

godalming123 committed:
fix bug

godalming123 committed:
fix bugs

godalming123 committed:
try and fix bugs

godalming123 committed:
fix bugs

godalming123 committed:
test light theme

godalming123 committed:
fix ptint media on dark mode

godalming123 committed:
change color-schem to auto

godalming123 committed:
test

godalming123 committed:
try and fix theme not deafualting to auto

godalming123 committed:
fix bug

godalming123 committed:
add back the color scheme definition in config so it works

godalming123 commented:
**Update:**
I have a working branch in https://github.com/godalming123/minimal/tree/support-for-changing-theme-in-_config.yaml however the only problem is it doesn't deafualt to auto theme so if you do not set `color-scheme: dark | light | auto | auto-deafualt-dark` in config.yaml then the document will have no colors. Currently the github pages instance is based on this branch so just go to https://godalming123.github.io/minimal/ to see working copy.

godalming123 committed:
add support for a color scheme that deafualts to dark if the device does not support themeing based on device theme

godalming123 committed:
fix mistype

godalming123 committed:
try and make color-scheme deafualt to auto

godalming123 committed:
try and make it work with blank color-scheme in config.yaml

godalming123 committed:
see if new approach still works when color-scheme is defined

godalming123 committed:
try deferent approch

godalming123 committed:
try more types

godalming123 committed:
Merge branch 'master' into support-for-changing-theme-in-_config.yaml

godalming123 committed:
Merge pull request #1 from godalming123/support-for-changing-theme-in-_config.yaml

Support for changing theme in  config.yaml

godalming123 commented:
just added support for changing the theme with a config.yaml option that deafualts to auto in master brach!

godalming123 committed:
remove unnesersary line

BDHU commented on _layouts/default.html:
```diff
@@ -6,7 +13,8 @@
 
 {% seo %}
-    <link rel="stylesheet" href="{{ "/assets/css/style.css?v=" | append: site.github.build_revision | relative_url }}">
+    <link rel="stylesheet" href="{{ "/assets/css/colors-ColorScheme.css?v=" | append: site.github.build_revision | relative_url | replace: "ColorScheme", {{ColorScheme}} }}">
```
I kept getting Liquid syntax error warning for this specific line. Perhaps remove the {{ for the colorScheme variable?

godalming123 replied:
will use your suggestions sorry for being slow lets hope it compiles without error 🤞

godalming123 replied:
the build worked! thanks for feedback

godalming123 committed:
use bdhu suggestions

ElManouche commented on README.md:
````diff
@@ -40,6 +40,7 @@ Additionally, you may choose to set the following optional variables:
 ```yml
 show_downloads: ["true" or "false" (unquoted) to indicate whether to provide a download URL]
 google_analytics: [Your Google Analytics tracking ID]
+color-scheme: ["dark", "light", "auto" or "auto-deafualt-dark" auto is deafault setting and changes theme based on device theme, auto-deafualt-dark is the same except if you device does not support changing theme based on device theme it will deafault to dark and the others are fairly explanetory.]
````
typo on "default"

godalming123 replied:
fixed

ElManouche reviewed (approved):
Good job

godalming123 committed:
fix spelling and grammer

davorpa commented on _config.yml:
```diff
@@ -4,3 +4,4 @@ description: Minimal is a theme for GitHub Pages.
 show_downloads: true
 google_analytics:
 theme: jekyll-theme-minimal
+#color-scheme: auto-deafualt-dark
```
Are there a typo `deafualt` according to README.md? I think this would be `default`. Renaming files are also needed

```suggestion
#color-scheme: auto-default-dark
```

davorpa commented on _layouts/default.html:
```diff
@@ -6,7 +13,8 @@
 
 {% seo %}
-    <link rel="stylesheet" href="{{ "/assets/css/style.css?v=" | append: site.github.build_revision | relative_url }}">
+    <link rel="stylesheet" href="{{ "/assets/css/colors-ColorScheme.css?v=" | append: site.github.build_revision | relative_url | replace: "ColorScheme", ColorScheme }}">
```
It's better place the replace filter at the beginning. Only thus, is possible avoid conflicts with the others

```suggestion
    <link rel="stylesheet" href="{{ "/assets/css/colors-ColorScheme.css?v=" | replace: "ColorScheme", ColorScheme | append: site.github.build_revision | relative_url }}">
```

godalming123 replied:
Why is it better putting the replace before?

godalming123 replied:
Also thanks for helping me develop the fork it has been much appreciated

davorpa replied:
> Why is it better putting the replace before?

mainly because filters are chained. 

Imagine that after process `relative_url`, the string contains `ColorScheme` (eg. `http://host/ColorScheme/assets/css/colors-ColorScheme.css` blah blah...). Then, with the replacement, the css file doesn't load, raising a 404 - Not Found 😉.

ldemailly commented:
It'd be great to see this merged but in meantime, what do I put in my `_config.yml` to use this branch/fork?

davorpa commented:
> It'd be great to see this merged but in meantime, what do I put in my `_config.yml` to use this branch/fork?

See how to `remote-theme` gem works. can be pointed to any repo+branch

https://github.com/benbalter/jekyll-remote-theme

godalming123 committed:
update usage instructions

godalming123 committed:
fix bugs

godalming123 committed:
Update _config.yml

Co-authored-by: David Ordás <3125580+davorpa@users.noreply.github.com>

godalming123 commented:
> It'd be great to see this merged but in meantime, what do I put in my `_config.yml` to use this branch/fork?

Sorry for not looking at the comments for a while to use this theme follow the instructions at https://github.com/godalming123/minimal#usage and then deploy your site with github pages

godalming123 committed:
fix spelling errer

godalming123 committed:
move replace statement forward

Co-authored-by: David Ordás <3125580+davorpa@users.noreply.github.com>

davorpa commented:
@godalming123, to be merged as feature in  github-pages you should revert changes made in theme name:
- 3c60bc02c05ceae497283157bd27b7650ab265fa
- 4c92f900d46493e4aa912c8fe855924b5d8a7fb2

I highly recommend using feature branches workflow instead coding all in `master/main` 🤗

In any way.... **great work**!!!!

godalming123 commented:
> @godalming123, to be merged as feature in github-pages you should revert changes made in theme name:
> 
> * [3c60bc0](https://github.com/pages-themes/minimal/commit/3c60bc02c05ceae497283157bd27b7650ab265fa)
> * [4c92f90](https://github.com/pages-themes/minimal/commit/4c92f900d46493e4aa912c8fe855924b5d8a7fb2)
> 
> I highly recommend using feature branches workflow instead coding all in `master/main` hugs
> 
> In any way.... **great work**!!!!

I'm probably not going to add this until maintainers see the pull request as my current goal would be to attract as many new users and that means documenting how to set this up so they can use it. And I dont even know if this repo is still maintained.

godalming123 commented:
# Want a more maintained fork?
If you would like a better maintained fork (as I rarely see posts here and am no longer updating this pull request as I see it as complete) see: https://github.com/BDHU/minimalist

davorpa commented:
> and that means documenting how to set this up so they can use it. And I dont even know if this repo is still maintained.

@parkr, as ex-github... do you know if `pages-themes` project is carried on maintained?

godalming123 committed:
fix preview link

godalming123 committed:
fix some awful style choices
